sctp/src/queue/pending_queue.rs

```diff
@@ -23,6 +27,42 @@ pub(crate) struct PendingQueue {
     unordered_is_selected: AtomicBool,
 }
 
+/// Simple diy semaphore not directly protecting a resource other than the current capacity of the Queue in bytes
+#[derive(Debug)]
+struct PushLimitSemaphore {
```

Comment:
`tokio::sync::Sempahore` instead?

Reply:
The stream is in sync code, I don't think using the tokio semaphore is a good choice there?

Reply:
See comment below